android/examples/unrooted_android.c

```diff
@@ -0,0 +1,300 @@
+/*
+ *  libusb example program for reading out the usb device specifications for Unrooted Android Devices based of testlibusb.c
+ *  Copyright 2020 Peter Stoiber
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ *  Please contact the author if you need another license.
+ *  This Repository is provided "as is", without warranties of any kind.
+*/
+
+/*
+ * This example creates a shared Object which can be accessed over JNA or JNI from Java or Kotlin in Android.
+ * Hint: If you are using Android Studio set the "Debug type" to "Java Only" to receive debug messages. 
+ */
+
+/*
+ * Todo:
+ * First you have to connect your Usb Device from the Java side.
+ * Use the android.hardware.usb class for finding the USB Device, claiming the interfaces and open the usb_device_connection.
+ * Obtain the native File Descriptor --> usb_device_connection.getFileDescriptor()
+ * Pass the received int Value to the unrooted_usb_description method of this code (over JNA)
+ */
+
+/*
+ * LibUsb can only be included to Android Projects Source over NDK up to now. (CMake is not supported at the moment)
+ * (Android Studio)
+ * Clone the last LibUsb GitHub Repo to your Android Project and include the Android.mk file to your build.gradle.
+ */
+
+/*
+ Example JNA Approach:
+    public interface unrooted_sample extends Library {
+        public static final unrooted_sample INSTANCE = Native.load("unrooted_android", unrooted_sample.class);
+        public int main (int fileDescriptor);
```

Comment:
Hi, this function should be renamed into unrooted_usb_description since e84f30f
This is a confusion mentioned here https://github.com/libusb/libusb/issues/717#issuecomment-753718501 

And It's not consistent with android/README of this PR https://github.com/libusb/libusb/pull/830/files#diff-16202c3941de7d6c550a2d075887e6c29f5ea083da72f53afd72ffc540687b95R71 

By the way, these examples deserve to be more complete rather than hints.
Maybe a link to https://github.com/Peter-St/unRootedSample (updated with the same names) ?

The same goes for #874 , without proper/full example it is difficult to test the code easily especially when you are not familiar with Android dev ;)

Reply:
right, I was too fast with my commits and forgot to rename this function.

About the links to complete repos outside of the libusb site I agree with you, that this would be easier for unfamiliar people, but up to now, no links for the other system platforms were given on this site. 
My unRootedSample project was a guidance for a single person. It was not designed for a general assistance scheme.